modules/ROOT/pages/dialog-components.adoc

```diff
@@ -539,6 +539,96 @@ A *textarea* is a multiline text field.
 }
 ----
 
+[[tree]]
+=== tree
+
+A *tree* is a composite component that presents an expandable category tree expandable down one level. It can also include a search and filter field.
+
+*onLeafAction:* Interacting with a *tree* component executes the `+onLeafAction+` callback set in the dialog configuration.
+
+[source,js]
+----
+tinymce.init({
+  selector: "textarea",
+  toolbar: "tree",
+  setup: (ed) => {
+    ed.ui.registry.addButton('tree', {
+      text: 'Tree toolbar button',
+      onAction: () => {
+        ed.windowManager.open({
+          title: 'Tree dialog',
+          buttons: [{type: 'cancel', text: 'Cancel',}],
+          body : {
+            type: 'panel',
+            items: [{
+              type: 'tree',
+              onLeafAction: (id) => {
+                console.log('clicked on item with id', id);
+              },
+              items: [
+                {
+                  type: 'directory',
+                  id: 'dir',
+                  title: 'Directory',
+                  menu: {
+                    icon: 'image-options',
+                    type: 'menubutton',
+                    fetch: (success) => success([
+                      {
+                        type: 'menuitem',
+                        text: 'Menu item',
+                        onAction: () => {
+                          console.log; // for demo purposes
+                        }
+                      }
+                    ])
```

Comment:
```suggestion
                    fetch: (success) => success([
		     'bold', 'italic', 'underline'
                    ])
```
These are menu items, so some that already exist can be called in this fetch function.